packages/front-end/enterprise/components/Dashboards/DashboardEditor/index.tsx

```diff
@@ -0,0 +1,486 @@
+      })
+    );
+    setAddBlockIndex(index);
+  };
+
+  if (blocks.length === 0 && !isDefined(addBlockIndex)) {
+    return (
+      <div className="mt-3">
+        <Flex
+          direction="column"
+          align="center"
+          justify="center"
+          px="80px"
+          pt="60px"
+          pb="70px"
+          className="appbox"
+          gap="5"
+        >
+          <Flex direction="column">
+            <Heading weight="medium" align="center">
+              Build a Custom Dashboard
+            </Heading>
+            <Text align="center">
+              Choose a block type to get started. Rearrange blocks to tell a
+              story with experiment data.
+            </Text>
+          </Flex>
+          {canEdit && (
+            <AddBlockDropdown
+              addBlockType={addBlockType}
+              trigger={
+                <Button icon={<PiCaretDownFill />} iconPosition="right">
+                  Add block
+                </Button>
+              }
+              forceToEditing={forceToEditing}
+            />
+          )}
+        </Flex>
+      </div>
+    );
+  }
+
+  const renderSingleBlock = ({
+    i,
+    key,
+    block,
+    setBlock,
+    isEditingBlock,
+  }: {
+    i: number | undefined;
+    key: number | string;
+    block: DashboardBlockInterfaceOrData<DashboardBlockInterface>;
+    setBlock: React.Dispatch<
+      DashboardBlockInterfaceOrData<DashboardBlockInterface>
+    >;
+    isEditingBlock: boolean;
+  }) => {
+    return (
+      <Flex
+        direction="column"
+        key={key}
+        ref={(el) => (isDefined(i) ? (blockRefs.current[i] = el) : null)}
+      >
+        <DashboardBlock
+          block={block}
+          dashboardExperiment={experiment}
+          isEditing={isEditing}
+          editingBlock={isEditingBlock}
+          disableBlock={editDrawerOpen && !isEditingBlock}
+          isFirstBlock={i === 0}
+          isLastBlock={i === blocks.length - 1}
+          setBlock={setBlock}
+          editBlock={() => {
+            setEditingBlockIndex(i);
+          }}
+          duplicateBlock={() => {
+            if (isDefined(i)) {
+              setAddBlockIndex(i + 1);
+              setStagedAddBlock(getBlockData(block));
+            }
+          }}
+          deleteBlock={() => {
+            if (isDefined(i)) {
+              setBlocks([...blocks.slice(0, i), ...blocks.slice(i + 1)]);
+            }
+          }}
+          moveBlock={(direction) => {
+            if (isDefined(i)) {
+              const otherBlocks = blocks.toSpliced(i, 1);
+              setBlocks([
+                ...otherBlocks.slice(0, i + direction),
+                block,
+                ...otherBlocks.slice(i + direction),
+              ]);
+            }
+          }}
+          mutate={mutate}
+        />
+        {isEditing && (
+          <Flex justify="center" mb="1em" position="relative">
+            {(hoverAddBlock === i || addBlockDropdown === i) && (
+              <div
```

Comment:
nit ui: It is not centered / it gets rendered over the + button

<img width="84" height="73" alt="Screenshot 2025-07-30 at 4 22 23 PM" src="https://github.com/user-attachments/assets/fe3ada2d-4740-4bbf-8349-f769e80762a2" />

Reply:
Also the + is not centered in the box 👀 
<img width="145" height="100" alt="Screenshot 2025-07-31 at 8 48 53 AM" src="https://github.com/user-attachments/assets/e8b36664-92cc-47c6-a4ac-955f482e9ba0" />